compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -247,15 +214,17 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
                                 true
                             }
                             VarValue::Value(cur_region) => {
-                                let lub = match self.lub_empty(cur_region) {
-                                    Ok(r) => r,
+                                let lub = match *cur_region {
                                     // If the empty and placeholder regions are in the same universe,
                                     // then the LUB is the Placeholder region (which is the cur_region).
                                     // If they are not in the same universe, the LUB is the Static lifetime.
-                                    Err(placeholder) if a_universe == placeholder.universe => {
-                                        cur_region
+                                    RePlaceholder(placeholder)
+                                        if a_universe != placeholder.universe =>
```

Comment:
I wonder why this isn't doing the same thing as below where it's checking that it can be named, rather than just checking if the universes are equal

Reply:
Ohh right, that's pre-existing. Let me try to make that change.